src/erlcloud_ses.erl

```diff
@@ -1093,6 +1160,12 @@ encode_destination(ToAddress, Acc) when is_list(ToAddress); is_binary(ToAddress)
     %% Single entry
     encode_destination_pairs([{to_addresses, [ToAddress]}], Acc).
 
+encode_destinations({destination_addresses, Destinations}, Acc) ->
```

Comment:
why `destination_addresses` and not `destinations`?

Reply:
I have `destinations` as the top level option and used `destination_addresses` to be similar to `to_addresses, cc_addresses, bcc_addresses` in the send_email functionality.

I could rework the function to accept either a single email address or a list of email addresses passed in via the tuple `{destinations, [emails()]}`. The main issues I am trying to solve for is knowing a list of addresses vs a string address.

Reply:
The issues is [here](https://github.com/erlcloud/erlcloud/pull/657/files#diff-c45cdf8adff42f1b80f94fd8c2e2be6eR742-R745).
```-type send_raw_email_destinations() :: email() | {destination_addresses, emails()}.

-type send_raw_email_opt() :: {source, email()} |
                              {destinations, send_raw_email_destinations()}.
```
supporting `send_raw_email_destinations()` to be `email() | {destination_addresses, emails()}` or `email() | {destination_addresses, [emails()]}`

Reply:
I get that `email()` can be string or binary, which sucks, but it's not too hard to pattern match string/binary vs. list of string/binary:

```erlang
encode_destinations([Dest|_T] = Destinations, Acc) when is_list(Dest); is_binary(Dest) ->
    encode_list("Destinations", Destinations, Acc);
encode_destinations(ToAddress, Acc) when is_list(ToAddress); is_binary(ToAddress) ->
    encode_destinations([ToAddress], Acc).
```

Seems like it's just awkward to write `[{destinations, {destination_addresses, [...]}}]` when it's pretty straightforward to discriminate between the 2 classes. If we supported `iolist()` that would be another story, but I don't think there is a need there.